src/solver/summary.h

```diff
@@ -53,6 +54,14 @@ class summaryt
   predicatet bw_transformer; // backward summary (over- or under-approx)
   predicatet bw_invariant; // backward invariant (over- or under-approx)
 
+  // The following fields are used for GOTO code instrumentation. It is
+  // necessary to transfer domain and value ownership to summary so that
+  // code instrumentation can obtain guards and invariants to be used.
```

Comment:
Let's make the comment less specific to GOTO instrumentation:
```
// The following fields represent the used abstract domains (containing 
// the template) and the corresponding computed abstract values.
// These can be useful to retrieve information that cannot be easily parsed 
// from the invariant expression.
```

Reply:
Changed. I recently realised that storing only a single pointer to domain may be a problem. There is only one summary for each function and when adding a summary to `summary_db`, if a summary for the given function already exists, the summaries are joint. Merging two different domains and values probably won't be possible, will it? So my line of thought is that we may need to store a vector of domains/values

Reply:
I'd stick with one summary per function. If one needs multiple summaries in different domains for a single function, where the result summary is a conjunction of the individual summaries, it's possible to use `product_domain`. If a different composition of summaries should be done, I think that a new combination domain should be created.